From 91a8092bf39709c48161268cc593bf950ea2fa7c Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Mon, 3 Apr 2023 10:03:57 +0200
Subject: [PATCH 114/162] fix(LinstorSR): retry if we can't build volume cache

Otherwise after SR creation, the master PBD can be unplugged.
See: https://xcp-ng.org/forum/post/60726

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/LinstorSR.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/LinstorSR.py b/drivers/LinstorSR.py
index 48feec7..324033a 100755
--- a/drivers/LinstorSR.py
+++ b/drivers/LinstorSR.py
@@ -1553,7 +1553,11 @@ class LinstorSR(SR.SR):
     def _create_linstor_cache(self):
         self._all_volume_metadata_cache = \
             self._linstor.get_volumes_with_metadata()
-        self._all_volume_info_cache = self._linstor.get_volumes_with_info()
+        self._all_volume_info_cache = util.retry(
+            self._linstor.get_volumes_with_info,
+            maxretry=10,
+            period=1
+        )
 
     def _destroy_linstor_cache(self):
         self._all_volume_info_cache = None
-- 
2.43.0

